<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
import pandas as pd

# Membuat dataframes dari dictionary
d = {'A':[1, 2, 3], 'B':[4, 5, 6],'C':[7, 8, 9]}

df = pd.DataFrame(d)
df


A  B  C
0  1  4  7
1  2  5  8
2  3  6  9

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [2]:
# Membaca data dan memasukannya kedalam suatu variabel
df = pd.read_csv('https://github.com/tidyverse/ggplot2/raw/master/data-raw/diamonds.csv')
df

carat        cut color clarity  depth  table  price     x     y     z
0       0.23      Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1       0.21    Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2       0.23       Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3       0.29    Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4       0.31       Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75
...      ...        ...   ...     ...    ...    ...    ...   ...   ...   ...
53935   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53936   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53937   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53938   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53939   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

[53940 rows x 10 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:
- Series hanya dapat berisi satu tipe data, sedangkan list dan dictionary tidak
- Dataframe 2D, Series 1D

In [3]:
# code here
a= [1, 2, 3, 4]

b = pd.Series(a)
b

0    1
1    2
2    3
3    4
dtype: int64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [4]:
# code here
df

carat        cut color clarity  depth  table  price     x     y     z
0       0.23      Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1       0.21    Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2       0.23       Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3       0.29    Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4       0.31       Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75
...      ...        ...   ...     ...    ...    ...    ...   ...   ...   ...
53935   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53936   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53937   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53938   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53939   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

[53940 rows x 10 columns]

In [5]:
print(df.price[0])
print(df['color'][len(df)-1])

326
D


<h2>Soal 5: Statistika Deskriptif </h2>

Jelaskan apa itu statistika deskriptif!

Jawab:
Statistika deskriptif adalah metode-metode yang berkaitan dengan pengumpulan dan penyajian suatu gugus data sehingga memberikan informasi yang berguna. Pengklasifikasian menjadi statistika deskriptif dan statistika inferensi dilakukan berdasarkan aktivitas yang dilakukan

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [6]:
# code here
df.describe()

carat         depth  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean       0.797940     61.749405  ...      5.734526      3.538734
std        0.474011      1.432621  ...      1.142135      0.705699
min        0.200000     43.000000  ...      0.000000      0.000000
25%        0.400000     61.000000  ...      4.720000      2.910000
50%        0.700000     61.800000  ...      5.710000      3.530000
75%        1.040000     62.500000  ...      6.540000      4.040000
max        5.010000     79.000000  ...     58.900000     31.800000

[8 rows x 7 columns]

Jawab:
- rata2 column price = 3932.799722
- standar deviasi column depth = 1.432621
- maksimum column carat = 5.010000

---

## Update Corona

In [7]:
import requests
import pandas as pd
 
def getJSONData(apiURL):
   response = requests.get(
   apiURL,
   verify = True)
  
   response.raise_for_status()
 
   return(response.json())
  
apiURL = "https://api.kawalcorona.com/"
datakorona = getJSONData(apiURL)

b = []
for i in range(len(datakorona)):
  b.append(datakorona[i]['attributes'])

df = pd.DataFrame.from_dict(b)

df

OBJECTID         Country_Region    Last_Update  ...  Deaths  Recovered   Active
0          18                     US  1594024439000  ...  129947   906763.0  1902618
1          22                 Brazil  1594024439000  ...   64867  1029045.0   509143
2          27                  India  1594024439000  ...   19693   424433.0   253287
3          14                 Russia  1594024439000  ...   10271   453495.0   223011
4          23                   Peru  1594024439000  ...   10589   193957.0    98172
..        ...                    ...            ...  ...     ...        ...      ...
183       149  Saint Kitts and Nevis  1594024439000  ...       0       15.0        1
184        93               Holy See  1594024439000  ...       0       12.0        0
185       141       Papua New Guinea  1594024439000  ...       0        8.0        3
186       185         Western Sahara  1594024439000  ...       1        8.0        1
187       117             MS Zaandam  1594024439000  ...       2        NaN        7

[188 rows x 9 columns]

In [8]:
indo =df.loc[df['Country_Region'].values == 'Indonesia',:]
indo

OBJECTID Country_Region    Last_Update  ...  Deaths  Recovered  Active
25        96      Indonesia  1594024439000  ...    3171    29105.0   31473

[1 rows x 9 columns]

In [9]:
from datetime import datetime
a = indo['Last_Update'].values[0]/1000
print(f'Last update: {datetime.utcfromtimestamp(a)}')

Last update: 2020-07-06 08:33:59
